libexec/python/shub/api.py

```diff
@@ -80,6 +80,11 @@ def load_image(self, image):
                                      uri='shub://',
                                      default_registry=SHUB_API_BASE,
                                      quiet=True)
+        # parse_image_uri may return an empty namespace cause that's allowed
+        # with docker://, but not with shub://
+        if len(self.image["namespace"]) == 0:
+            bot.error("Namespace cannot be empty for shub:// url!")
+            sys.exit(1)
 
```

Comment:
What would be an example image name to produce an empty namespace? I am trying (and failing) to get one that doesn't result in `library`

Reply:
Great, it means the parsing code is working! In an earlier (uncommited) version, I just made namespaces optional, without distinguishing between docker:// and shub://, so this check was needed.
Once I switched to using regexes, those *should* already enforce that, for shub://, the namespace is never empty, so this should be changed into an assert instead (or whichever is the preferred way to check for impossible conditions).

Reply:
okay, excellent.